packages/aws-client/lib/S3MultipartUploads.js

```diff
@@ -0,0 +1,115 @@
+// Utility functions to help with S3 multi-part uploads
+
+'use strict';
+
+const range = require('lodash/range');
+const { s3 } = require('../services');
+
+const GB = 1024 * 1024 * 1024;
+
+// Each part of a multi-part copy needs to specify a byte range to be copied.
+// This byte range has a starting byte and an ending byte (inclusive) that makes
+// up the part. All parts must be larger than 5 MB, except that the last part
+// may be smaller than 5 MB. Parts cannot be larger than 5 GB.
+//
+// This function takes a file size and an optional maxSize. It returns an array
+// of objects, each containing a `start` and an `end` value. These will make up
+// the ranges of the multi-part copy.
+//
+// https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPartCopy.html
+const createMultipartChunks = (size, maxSize = 5 * GB) => {
+  const lastChunkSize = size % maxSize;
+
+  // Build the list of full-size chunks
+  const chunks = range(0, size - lastChunkSize, maxSize).map((start) => {
```

Comment:
nit: No need to subtract `lastChunkSize`

Reply:
👍 Will fix.

Reply:
Removing that does actually cause test failures:

```
lib › S3MultipartUploads › createMultipartChunks(11, 10) returns the correct chunks

  tests/lib/test-S3MultipartUploads.js:36

   35: test('createMultipartChunks(11, 10) returns the correct chunks', (t) => {
   36:   t.deepEqual(                                                           
   37:     createMultipartChunks(11, 10),                                       

  Difference:

    [
      {
        end: 9,
        start: 0,
      },
  -   {
  -     end: 19,
  -     start: 10,
  -   },
      {
        end: 10,
        start: 10,
      },
    ]

  › test (tests/lib/test-S3MultipartUploads.js:36:5)
  › process._tickCallback (internal/process/next_tick.js:68:7)
```

Reply:
Yes, that's because the next line should be corrected from this:
```
const end = start + maxSize - 1;
```
to this:
```
const end = Math.min(start + maxSize, size) - 1;
```
which would then allow you to eliminate the block for creating the last chunk that follows.

Reply:
I agree that would work. My concern is that this function is a little confusing already. I'd rather err on the side of being a little more verbose so that, when someone has to fix a bug in it on a Friday night after a couple 🍺 🍺 , it'll be easier to understand.